src/utils/database/weekly_challenge_table.py

```diff
@@ -0,0 +1,47 @@
+"""
+Weekly Challenge table module
+"""
+
+from tinydb import TinyDB, where
+from .db_decorators import validate_insert
+
+class WeeklyChallengeTable():
+    """
+    Table to hold all Weekly Challenges after being generated
+    """
+    TABLE_WEEKLY_CHALLENGE_FIELDS = ["id", "date"]
+    TABLE_WEEKLY_CHALLENGE = "Weekly_Challenge"
```

Comment:
move to __init__() as self.table_fields and self.table_name

Reply:
Renamed but kept outside of constructor due to issues with the validate_insert decorator.